wavelength-plugins/tsumeta.js

```diff
@@ -0,0 +1,205 @@
+}
+
+function isTsuMetaCouncil(user) {
+	if (!user) return;
+	if (typeof user === 'object') user = user.userid;
+	let council = Db.councilmember.get(toId(user));
+	if (council === 1) return true;
+	//Denies being "a TsuMeta Council Member" in proposals and such when suspended
+	return false;
+}
+
+exports.commands = {
+	tsucouncil: "tsumeta",
+	tsumembers: "tsumeta",
+	tsumetacouncil: "tsumeta",
+	tsumeta: {
+		give: function (target, room, user) {
+			if (user.userid !== "desokoro") return this.errorReply("You must be Desokoro to add users into the council.");
+			if (!target) return this.parse("/tsumetahelp");
+			let tsuMetaMember = toId(target);
+			if (tsuMetaMember.length > 18) return this.errorReply("Usernames cannot exceed 18 characters.");
+			if (isTsuMetaCouncil(tsuMetaMember)) return this.errorReply(`${tsuMetaMember} is already in the TsuMeta Council.`);
+			Db.councilmember.set(tsuMetaMember, 1);
+			this.sendReply(`|html|${WL.nameColor(tsuMetaMember, true)} has been successfully been added into the TsuMeta Council.`);
+			if (Users.get(tsuMetaMember)) Users(tsuMetaMember).popup(`|html|You have been added into the TsuMeta Council by ${WL.nameColor(user.name, true)}.`);
+		},
+
+		kick: "take",
+		remove: "take",
+		delete: "take",
+		take: function (target, room, user) {
+			if (user.userid !== "desokoro") return this.errorReply("You must be Desokoro to remove users into the council.");
+			if (!target) return this.parse(`/tsumetahelp`);
+			let tsuMetaMember = toId(target);
+			if (tsuMetaMember.length > 18) return this.errorReply("Usernames cannot exceed 18 characters.");
+			if (!isTsuMetaCouncil(tsuMetaMember)) return this.errorReply(`${tsuMetaMember} isn't a TsuMeta council member.`);
+			Db.councilmember.remove(tsuMetaMember);
+			this.sendReply(`|html|${WL.nameColor(tsuMetaMember, true)} has been removed from the TsuMeta council.`);
+			if (Users.get(tsuMetaMember)) Users(tsuMetaMember).popup(`|html|You have been removed from the TsuMeta Council by ${WL.nameColor(user.name, true)}.`);
+		},
+
+		users: 'list',
+		list: function (target, room, user) {
+			if (!Db.councilmember.keys().length) return this.errorReply('There seems to be no users in the TsuMeta Council.');
+			let display = [];
+			Db.councilmember.keys().forEach(councilMember => {
+				display.push(WL.nameColor(councilMember, (Users(councilMember) && Users(councilMember).connected)));
+			});
+			this.popupReply(`|html|<strong><u><font size="3"><center>TsuMeta Council Members:</center></font></u></strong>${display.join(',')}`);
+		},
+
+		meeting: "message",
+		alert: "message",
+		pm: "message",
+		message: function (target, room, user) {
+			if (user.userid !== "desokoro" && !this.can("bypassall")) return this.errorReply(`This command is reserved for Desokoro only.`);
+			if (!target) return this.parse("/tsumetahelp");
+			let councilMembers = Db.councilmember.keys();
+			for (let u in councilMembers) {
+				if (!Users(councilMembers[u]).connected) continue;
```

Comment:
make sure the user object exists before checking if the user is connected

Reply:
how?

Reply:
`if (!Users(councilMembers[u]) || !Users(councilMembers[u]).connected) continue;`